src/schema/validator.js

```diff
@@ -25,11 +25,51 @@ import schemas from './imported';
 
 const jsonSchemaDraft06 = require('ajv/lib/refs/json-schema-draft-06');
 
-function filterErrors(errors) {
+function filterErrors(errors, manifestData = undefined) {
   if (errors) {
-    return errors.filter((error) => {
+    let filteredErrors = errors.filter((error) => {
       return error.keyword !== '$merge';
     });
+
+    // Filter out errors related to future manifest versions.
+    if (
+      filteredErrors.length > 0 &&
+      manifestData?.manifest_version === MANIFEST_VERSION_DEFAULT
```

Comment:
shouldn't we use min/max versions here and below instead of the default version? We only filter out eror for "future manifest versions" because the default is set to `2` but would it make sense to filter out errors that are outside the range specific by the user?

Reply:
@willdurand and I discussed about this comment in a meeting, and this was a summary of what we discussed:
- the approach in this PR aims to only report errors that are relevant for the manifest_version of the addon being validated ([1])
- the reasoning behind this approach is that:
  - the extension developers would not get wrong the manifest_version, and so the validation errors should only suggest the changes that are relevant to make the addon valid with the manifest_version set in their manifest
  - if an extension developer wants to port its extension to manifest_version 3, they would be changing the manifest_version set in their extension manifest, and at that point the errors related to that schema versions becomes relevant  (and the ones related to the older manifest version are not relevant anymore)
  - the addons-linter cli options --min-manifest-version / --max-manifest-version are meant to allow the developers to set the manifest_version to a value that is not yet allowed by AMO, but it should not change the errors that we are reporting (until the developer decide to actually set a different manifest_version on the extension manifest)

----
[1]: the new version of this patch, which the additions from 17eea76, does not use MANIFEST_VERSION_DEFAULT in these checks anymore, it does use the manifestData.manifest_version instead